### PR TITLE
fix lines, linesegments, for single and multicolor drawing

### PR DESCRIPTION
**Pull request by jkrumbiegel**

### CODE REVIEW

**Comment by asinghvi17** [comment]
Looking at the test logs, one thing we still have to account for is variable linewidths.  I still don't know what the best way to do that is - 4 methods to just draw lines seems a bit excessive.

File: src/CairoMakie.jl [hunk 1]
```diff
@@ -249,17 +209,111 @@ function draw_atomic(scene::Scene, screen::CairoScreen, primitive::Union{Lines,
     ctx = screen.context
     model = primitive[:model][]
     positions = primitive[1][]
+
     isempty(positions) && return
-    N = length(positions)
+
+    projected_positions = project_position.(Ref(scene), positions, Ref(model))
```
Comment (asinghvi17) [inline src/CairoMakie.jl:215]: Test failure happens here...specifically when broadcasting.  Will investigate further.

Comment (jkrumbiegel) [reply src/CairoMakie.jl:215]: ok this happens because this particular wireframe call results in `positions` being a view of tuples of points

Comment (jkrumbiegel) [reply src/CairoMakie.jl:230]: ok I've introduced a workaround for that case, now the tests fail only in reference image comparisons (which I guess doesn't mean much as it was bad before?)

Comment (asinghvi17) [reply src/CairoMakie.jl:230]: Ah, that's pretty annoying...guess it makes sense, though.  I guess we should handle that here as well?

Comment (asinghvi17) [reply src/CairoMakie.jl:230]: Yeah, the refimages are expected to fail.  I wonder if we could disable that...

**Pull request merged**